contracts/mainnet/connectors/yearn_v2/main.sol

```diff
@@ -0,0 +1,90 @@
+pragma solidity ^0.7.0;
+
+/**
+ * @title Yearn V2.
+ * @dev Vaults & yield.
+ */
+
+import { TokenInterface } from "../../common/interfaces.sol";
+import { Basic } from "../../common/basic.sol";
+import { Events } from "./events.sol";
+import { YearnV2Interface } from "./interface.sol";
+
+abstract contract YearnResolver is Events, Basic {
+    /**
+     * @dev Deposit funds in the vault, issuing shares to recipient.
+     * @notice This will deposit funds to a specific Yearn Vault.
+     * @param token The address of the token to deposit.
+     * @param vault The address of the vault to deposit funds into.
+     * @param amt The amount of tokens to deposit.
+     * @param getId ID to retrieve amt.
+     * @param setId ID stores the amount of shares received.
+    */
+    function deposit(
+        address token,
```

Comment:
@TBouder Just curious to know, why the `token` param is needed? We can simply apply the logics using `want`?